### PR TITLE
Add PCC check to text demo

### PR DESCRIPTION
**Pull request by alingTT**
### Problem description
Added PCC check to tt output with reference model output during 1 Layer text demo in TG llama.
Currrently we do not perform any PCC check to verify correctness of outputs.

### What's changed
There are 3 main PCC check tests can be ran from `text_demo.py`
1. Disabled (No PCC check) by running any test other than  id `batch-32-pcc`
2. Run a PCC test on 1L with randomized weights, this test ONLY compares PCC of the output of prefill btw tt model and reference model, not decode (since 1L case generally perform bad on PCC during decode), for this set `num_layers=1` and `instruct=False`
3. Run a PCC test on 80L with instruct weights, this test will compare PCC of output of prefill and decode. The output token will be swapped with output token from reference model via teacher forcing until `pcc_decode_len` is complete, for this set `num_layers=80` and `instruct=True`

### Note: 
- All PCC tests use "This is a test" as a prompt
- All tests pass locally 1L with no instruct with PCC, 80L with instruct with PCC. 
- Once reference logits are copied to CI machines, I will start a test on CI to verify test correctness

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable)
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable) TG Demo and TG Quick [Tests](https://github.com/tenstorrent/tt-metal/actions/runs/15784838785/job/44499126482) Passing
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable) See [recommended dev flow](https://github.com/tenstorrent/tt-metal/blob/main/models/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: models/demos/llama3_subdevices/demo/text_demo.py [hunk 1]
```diff
@@ -547,12 +629,13 @@ def test_demo_text(
 
             # Run decode forward
             try:
-                out_tok_cpu = generator.decode_forward_text(
+                tt_out_logits, tt_out_tok = generator.decode_forward_text(
                     out_tok,
                     current_pos,
-                    enable_trace=enable_trace,
+                    enable_trace=False,
```
Comment (sraizada-tt) [inline models/demos/llama3_subdevices/demo/text_demo.py]: we can't disable trace for all tests

File: models/demos/llama3_subdevices/tt/generator.py [hunk 2]
```diff
@@ -138,8 +140,7 @@ def prefill_forward_text(
                 self.seq_groups_to_batch_slot[s] = recently_filled_slots.pop(0)
 
         logger.info(f"Finished prefill for all users up to {batch_seq_len} tokens, Starting decode...")
-
-        return output_logits
+        return output_toks, output_logits
```
Comment (sraizada-tt) [inline models/demos/llama3_subdevices/tt/generator.py]: we can't change the output signature of this function because vllm uses it

**Review by sraizada-tt — CHANGES_REQUESTED** [review]

File: models/demos/llama3_subdevices/demo/text_demo.py (hunk 1 above)
Comment (alingTT) [reply models/demos/llama3_subdevices/demo/text_demo.py]: Fixed, only PCC check would disable trace

File: models/demos/llama3_subdevices/tt/generator.py (hunk 2 above)
Comment (alingTT) [reply models/demos/llama3_subdevices/tt/generator.py]: Fixed, function signature remains the same, logits are saved with in place logits tensor passed from `text_demo.py` into both `generator.prefill_forward_text()` and `generator.decode_forward_text()`.

File: models/demos/llama3_subdevices/demo/text_demo.py [hunk 3]
```diff
@@ -579,8 +709,10 @@ def test_demo_text(
 
             # Save output token to print out later
             for user in range(batch_size):
-                user_tok = tt_output_torch.tolist()[user]
-
+                if batch_size == 1:
```
Comment (sraizada-tt) [inline models/demos/llama3_subdevices/demo/text_demo.py]: this is the same thing, remove

**Review by sraizada-tt — APPROVED** [review]

File: tests/scripts/tg/run_tg_demo_tests.sh [hunk 4]
```diff
@@ -15,6 +15,7 @@ run_tg_llama3_tests() {
     LLAMA_DIR=$llama_dir TT_METAL_ENABLE_ERISC_IRAM=1 FAKE_DEVICE=TG pytest -n auto models/demos/llama3_subdevices/demo/text_demo.py -k "repeat" --timeout 1000; fail+=$?;
     LLAMA_DIR=$llama_dir TT_METAL_ENABLE_ERISC_IRAM=1 FAKE_DEVICE=TG pytest -n auto models/demos/llama3_subdevices/demo/text_demo.py -k "long-context-batch32" --timeout 1000; fail+=$?;
     LLAMA_DIR=$llama_dir TT_METAL_ENABLE_ERISC_IRAM=1 FAKE_DEVICE=TG pytest -n auto models/demos/llama3_subdevices/demo/text_demo.py -k "long-context-32k" --timeout 1000; fail+=$?;
+    LLAMA_DIR=$llama_dir TT_METAL_ENABLE_ERISC_IRAM=1 FAKE_DEVICE=TG pytest -n auto models/demos/llama3_subdevices/demo/text_demo.py -k "pcc-80L" --timeout 1000; fail+=$?;
```
Comment (roseli-TT) [inline tests/scripts/tg/run_tg_demo_tests.sh:18]: Does this test need timeout 1000? How long is the test expected to run for? Can it be shorter?

**Comment by djordje-tt** [comment]
Hey, can you take a look? @roseli-TT

File: tests/scripts/tg/run_tg_demo_tests.sh (hunk 4 above)
Comment (alingTT) [reply tests/scripts/tg/run_tg_demo_tests.sh:18]: this test should take the same/similar time as "long-context-batch32" the only difference is there is a teacher forcing check enforced to the outputs during decode (all other parameters are the same besides pcc_check and num_layers)

**Review by roseli-TT — APPROVED** [review]
